cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -280,7 +280,8 @@ public KafkaAssemblyOperator(Vertx vertx, PlatformFeaturesAvailability pfa,
                 .compose(state -> state.kafkaManualPodCleaning())
                 .compose(state -> state.kafkaNetPolicy())
                 .compose(state -> state.kafkaManualRollingUpdate())
-                .compose(state -> state.kafkaVersionChange())
+                .compose(state -> state.kafkaVersionChangeCheck())
+                .compose(state -> state.kafkaVersionChange(false))
```

Comment:
Do we want to use a boolean argument here, I wonder if it would be better to have two separate methods:
```
kafkaVersionChange
```
and
```
kafkaVersionChangePostponed
```

Reply:
The methods are otherwise almost the same. So is it really worth to have separate methods duplicating the code?

Reply:
I was thinking through this as I was reading through the method, it would be nice if they could be split, with the duplicated logic being moved out to separate methods, I started refactoring it locally and realized to rip out the boolean would be a non-trivial effort for not enough value. I thought it was worth bringing up, as boolean arguments normally can be extracted, but in this case it wouldn't be worth it.